Content.Client/StatusIcon/StatusIconOverlay.cs

```diff
@@ -72,9 +79,40 @@ protected override void Draw(in OverlayDrawArgs args)
             var accOffsetR = 0;
             icons.Sort();
 
+            Texture? texture = null;
             foreach (var proto in icons)
             {
-                var texture = _sprite.Frame0(proto.Icon);
+                switch (proto.Icon)
+                {
+                    case SpriteSpecifier.Rsi rsi:
+                        var rsiActual = _resourceCache.GetResource<RSIResource>(rsi.RsiPath).RSI;
+                        rsiActual.TryGetState(rsi.RsiState, out var state);
+                        var frames = state!.GetFrames(RSI.State.Direction.South);
+                        var delays = state.GetDelays();
+                        var totalDelay = delays.Sum();
+                        var time = _timing.RealTime.TotalSeconds % totalDelay;
+                        var delaySum = 0f;
+
+                        for (var i = 0; i < delays.Length; i++)
+                        {
+                            var delay = delays[i];
+                            delaySum += delay;
+
+                            if (time > delaySum)
+                                continue;
+
+                            texture = frames[i];
+                            break;
+                        }
+
+                        texture ??= _sprite.Frame0(proto.Icon);
+                        break;
+                    case SpriteSpecifier.Texture spriteTexture:
+                        texture = spriteTexture.GetTexture(_resourceCache);
+                        break;
+                    default:
+                        throw new NotImplementedException();
```

Comment:
Considering this code is shared with gastileoverlay it should be on a common system for handling it.